lib/introspector.js

```diff
@@ -45,42 +45,31 @@ const isNamedLine = line => NAMED_LINES.find(s => line.startsWith(s));
 const splitAtIndex = (line, index) => [line.slice(0, index), line.slice(index)];
 
 const splitComments = lines => {
-  let paramStart = lines.length;
-  let paramEnd = lines.length;
-  let paramsStarted = false;
-
-  for (let i = 1; i < lines.length; i++) {
-    if (!paramsStarted) {
-      if (
-        lines[i].startsWith(' '.repeat(2)) &&
-        !lines[i].startsWith(' '.repeat(4))
-      ) {
-        paramStart = i;
-        paramsStarted = true;
-      } else if (isNamedLine(lines[i])) {
-        paramStart = i;
-        paramEnd = i;
-        break;
-      }
-    } else if (paramsStarted && !lines[i].startsWith(' ')) {
-      paramEnd = i;
+  let i = 1;
+  let hasSignature = false;
+  while (i < lines.length) {
+    if (lines[i].startsWith(' ') || isNamedLine(lines[i])) break;
+    if (lines[i].startsWith('Signature:')) {
+      hasSignature = true;
       break;
     }
+    i++;
   }
+  const signatureStart = i;
 
-  let signatureStart = paramStart;
-
-  for (let i = 1; i < paramStart; i++) {
-    if (lines[i].startsWith('Signature:')) {
-      signatureStart = i;
-      break;
-    }
+  if (hasSignature) {
+    i++;
+    while (i < lines.length && lines[i].startsWith(' '.repeat(4))) i++;
```

Comment:
Perhaps extract `' '.repeat(4)` to a variable (or constant)?

Reply:
Done